src/core/symbology/qgsgraduatedsymbolrenderer.cpp

```diff
@@ -1108,6 +1204,25 @@ QDomElement QgsGraduatedSymbolRenderer::save( QDomDocument &doc, const QgsReadWr
     rendererElem.appendChild( modeElem );
   }
 
+  // symmetry
+  QDomElement symmetricModeElem = doc.createElement( QStringLiteral( "symmetricMode" ) );
+  symmetricModeElem.setAttribute( QStringLiteral( "enabled" ), mUseSymmetricMode ? QStringLiteral( "true" ) : QStringLiteral( "false" ) );
+  symmetricModeElem.setAttribute( QStringLiteral( "symmetryPoint" ), mSymmetryPoint );
+  symmetricModeElem.setAttribute( QStringLiteral( "astride" ), mAstride ? QStringLiteral( "true" ) : QStringLiteral( "false" ) );
+  if ( Pretty == mMode )
+  {
+    QString breaks;
+    for ( int i = 0; i < mListForCboPrettyBreaks.size() - 1; i++ ) // -1 to write 1/2/3 instead of 1/2/3/
+    {
+      breaks.append( mListForCboPrettyBreaks.at( i ) );
+      breaks.append( QStringLiteral( "/" ) );
+    }
+    breaks.append( mListForCboPrettyBreaks.at( mListForCboPrettyBreaks.size() - 1 ) );
```

Comment:
If the list is empty, this will crash. Probably responsible for the crash in the pull request.

Might be possible to use `QgsXmlTools::writeVariant( QVariant( mListForCboPrettyBreaks ) )` instead.